Tests/src/PublicAPITests/SequenceSystemTests.cpp

```diff
@@ -391,6 +432,46 @@ CSP_PUBLIC_TEST(CSPEngine, SequenceSystemTests, GetSequenceTest)
 }
 #endif
 
+#if RUN_ALL_UNIT_TESTS || RUN_SEQUENCESYSTEM_TESTS || RUN_SEQUENCESYSTEM_GETSEQUENCE_INVALIDKEY_TEST
+CSP_PUBLIC_TEST(CSPEngine, SequenceSystemTests, GetSequenceInvalidKeyTest)
+{
+	SetRandSeed();
+
+	auto& SystemsManager = csp::systems::SystemsManager::Get();
+	auto* UserSystem	 = SystemsManager.GetUserSystem();
+	auto* SpaceSystem	 = SystemsManager.GetSpaceSystem();
+	auto* SequenceSystem = SystemsManager.GetSequenceSystem();
+
+	// Log in
+	csp::common::String UserId;
+	LogIn(UserSystem, UserId);
+
+	// Create space
+	char UniqueSpaceName[256];
+	SPRINTF(UniqueSpaceName, "%s-%s", TestSpaceName, GetUniqueString().c_str());
+	const char* TestSpaceDescription = "CSP-UNITTEST-SPACEDESC-MAG";
+
+	csp::systems::Space Space;
+	CreateSpace(SpaceSystem, UniqueSpaceName, TestSpaceDescription, csp::systems::SpaceAttributes::Private, nullptr, nullptr, nullptr, Space);
+
+	// Create sequence
+	csp::common::Array<csp::common::String> SequenceItems {"Hotspot1", "Hotspot2", "Hotspot3"};
+	const char* TestSequenceKey = "CSP-UNITTEST/SEQUENCE-MAG";
+	char UniqueSequenceName[256];
+	SPRINTF(UniqueSequenceName, "%s-%s", TestSequenceKey, GetUniqueString().c_str());
+
+
+	csp::systems::Sequence RetrievedSequence;
+	GetSequence(SequenceSystem, UniqueSequenceName, RetrievedSequence, csp::systems::EResultCode::Failed);
```

Comment:
Won't this fail anyway as you haven't created a sequence to actually get?

Reply:
Here the comment says create sequence but it doesn't, it just creates an invalid name.

Reply:
Yes, it will indeed fail, but with a different error code: 
Attempting to get a non-existent sequence will result in a 404 error from CHS.
Attempting to get a sequence with an invalid character name (which is what we are testing for) will respond with an error code of 0. 

Thus, no sequence actually needs to be created to test that an invalid name is rejected with the correct error code. If you look at the call signature for GetSequence you will see that the default value for the expected return code is 0, so it's sort of implied.

That being said, it's only implied, and so not clear, so I will tidy up the comments and make it explicit.

Reply:
And this morning we agreed that you would add some additional comments to make it clear what was going on in each case.

Reply:
Done